store/etcdv3/workload.go

```diff
@@ -65,18 +65,18 @@ func (m *Mercury) GetWorkloadStatus(ctx context.Context, id string) (*types.Stat
 }
 
 // SetWorkloadStatus set workload status
-func (m *Mercury) SetWorkloadStatus(ctx context.Context, workload *types.Workload, ttl int64) error {
-	appname, entrypoint, _, err := utils.ParseWorkloadName(workload.Name)
-	if err != nil {
-		return err
+func (m *Mercury) SetWorkloadStatus(ctx context.Context, status *types.StatusMeta, ttl int64) error {
+	if status.Appname == "" || status.Entrypoint == "" || status.Nodename == "" {
+		return fmt.Errorf("SetWorkloadStatus error, status has no appname / entrypoint/ nodename")
```

Comment:
since we don't use `ParseWorkloadnName`, in order to guarantee the original behavior, we check appname, entrypoint and nodename here

Reply:
so there is nothing to format, why you use errorf here